src/utils/function_name.rs

```diff
@@ -0,0 +1,29 @@
+use crate::modules::types::Type;
+
+#[derive(Clone)]
+pub struct FunctionName {
+    name: String,
+    id: usize,
+    variant: usize,
+    returns: Type,
+}
+
+impl FunctionName {
+    pub fn new(name: &str, id: usize, variant: usize, returns: &Type) -> Self {
```

Comment:
The struct is pretty transparent, so I don't see a need for a construction method. Simply making all fields public, and using `FunctionName { ... }` construction should do.

Reply:
I always create a construction method for structs. In this case, it saves you doing the `String::from()` and `clone()` in the calling code, which would otherwise look like this:
```
meta.fun_name = Some(FunctionName { name: self.name.clone(), id: self.id, variant: index, returns: self.returns.clone() });
```